app/models/membership_application.rb

```diff
@@ -9,17 +9,115 @@ class MembershipApplication < ApplicationRecord
                         :last_name,
                         :company_number,
                         :contact_email,
-                        :status
+                        :state
+
   validates_length_of :company_number, is: 10
   validates_format_of :contact_email, with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: [:create, :update]
   validate :swedish_organisationsnummer
+
   accepts_nested_attributes_for :uploaded_files, allow_destroy: true
 
+
+  include AASM
+
+  aasm :column => 'state' do
+
+    state :pending, :initial => true
+    state :waiting_for_applicant
+    state :accepted
+    state :rejected
+
+
+    event :reject do
+      after do
+        reject_membership
+      end
+      transitions from: [:pending, :waiting_for_applicant, :accepted], to: :rejected
+    end
+
+    event :accept do
+      after do
+        accept_membership
+      end
+      # if a mistake was made and the application should have been accepted, can change it to accept
+      transitions from: [:pending, :rejected], to: :accepted, guard: [:paid?, :not_a_member?]
+    end
+
+    event :ask_applicant_for_info do
+      transitions from: [:pending, :rejected], to: :waiting_for_applicant, guard: :not_a_member?
+    end
+
+    event :cancel_waiting_for_applicant do
+      transitions from: [:waiting_for_applicant], to: :pending
+    end
+
+    event :applicant_updated_info do
+      transitions from: [:accepted, :rejected, :waiting_for_applicant, :pending], to: :pending, guard: :not_a_member?
+    end
+
+  end
```

Comment:
whew.  glad it was that simple.  Otherwise I was afraid there was something nasty we'd have to untangle. :-)